CI nix release check health failure: no input files

The CI Nix release `check-health` job fails after successfully building and testing the release.

```sh
Run JANK_SKIP_AOT_CHECK=1 nix run .?submodules=1 check-health
─ system ───────────────────────────────────────────────────────────────────────
─ ✅ operating system: linux
─ ✅ default triple: x86_64-pc-linux-gnu

─ jank install ─────────────────────────────────────────────────────────────────
─ ✅ jank version: jank-0.1-
─ ✅ jank resource dir: ../lib/jank/0.1  (found)
─ ✅ jank resolved resource dir: /nix/store/d84w29rb3icr8j8qa9r1gm8mbswng952-jank-git/bin/../lib/jank/0.1  (not found)
─ ✅ jank user cache dir: /home/runner/.cache/jank/x86_64-pc-linux-gnu-da02f099ea2882f9d19be4775ba9d14f0964ee6efb3ece0d36f6f4699ba8672c  (not found)

─ clang install ────────────────────────────────────────────────────────────────
─ ✅ configured clang path: /nix/store/j29kmh4xra206ysir0gdj4s7lk1bwlwv-llvm-jank-wrapper-21.0.0-git/bin/clang++ (found)
─ ✅ configured clang resource dir: /nix/store/76rnc1fr1f5qm1xzky2firs4bkrxkg70-llvm-jank-21.0.0-git/lib/clang/21 (found)

─ jank runtime ─────────────────────────────────────────────────────────────────
Note: Looks like your first run with these flags. Building pre-compiled header… failed!
─ internal/system-failure ──────────────────────────────────────────────────────
error: Clang failed with errors.error:  no such file or directory:              
       '<D0>M<U+0005>:AV'error:  no input files                                 
                                                                                

Error: Process completed with exit code 1.
```

Issue can't be reproduced locally

--- a/compiler+runtime/src/cpp/jank/aot/processor.cpp
+++ b/compiler+runtime/src/cpp/jank/aot/processor.cpp
@@ -139,8 +139,9 @@ int main(int argc, const char** argv)
 #endif
 
     std::filesystem::path const jank_path{ util::process_dir().c_str() };
+    std::string jank_path_str = jank_path.string();
     compiler_args.emplace_back(strdup("-L"));
-    compiler_args.emplace_back(strdup(jank_path.string().c_str()));
+    compiler_args.emplace_back(strdup(jank_path_str.c_str()));
 
     std::filesystem::path const jank_resource_dir{ util::resource_dir().c_str() };
     compiler_args.emplace_back(strdup("-I"));
@@ -363,11 +364,13 @@ int main(int argc, const char** argv)
     std::filesystem::path const jank_resource_dir{ util::resource_dir().c_str() };
     compiler_args.push_back("-include");
     auto const prelude_path{ jank_resource_dir / "include/cpp/jank/prelude.hpp" };
-    compiler_args.push_back(prelude_path.string().c_str());
+    std::string prelude_path_str = prelude_path.string();
+    std::string module_path_str = module_path.string();
+    compiler_args.push_back(prelude_path_str.c_str());
 
     compiler_args.push_back("-c");
     compiler_args.push_back("-o");
-    compiler_args.push_back(module_path.string().c_str());
+    compiler_args.push_back(module_path_str.c_str());
 
     compiler_args.push_back("-x");
     compiler_args.push_back("c++");

--- a/compiler+runtime/src/cpp/jank/environment/check_health.cpp
+++ b/compiler+runtime/src/cpp/jank/environment/check_health.cpp
@@ -407,11 +407,12 @@ namespace jank::environment
       aot_prc.build_executable(util::cli::opts.target_module).expect_ok();
 
       auto const stdout_file{ std::filesystem::path{ path_tmp } / "stdout" };
+      std::string stdout_file_str = stdout_file.string();
       auto const proc_code{ llvm::sys::ExecuteAndWait(
         exe_output.string(),
         { exe_output.string() },
         std::nullopt,
-        { std::nullopt, stdout_file.string().c_str(), std::nullopt },
+        { std::nullopt, stdout_file_str.c_str(), std::nullopt },
         5) };
       if(proc_code != 0)
       {

--- a/compiler+runtime/src/cpp/jank/util/clang.cpp
+++ b/compiler+runtime/src/cpp/jank/util/clang.cpp
@@ -91,7 +91,7 @@ namespace jank::util
     std::filesystem::path const configured_path{ JANK_CLANG_PATH };
     if(std::filesystem::exists(configured_path))
     {
-      return result = configured_path.string().c_str();
+      return result = configured_path.string();
     }
 
     std::filesystem::path const resource_dir{ util::resource_dir().c_str() };
@@ -278,6 +278,9 @@ namespace jank::util
     };
     std::filesystem::create_directories(output_path.parent_path());
 
+    std::string output_path_str = output_path.string();
+    std::string include_path_str = include_path.string();
+
     args.emplace_back("-Xclang");
     args.emplace_back("-fincremental-extensions");
     args.emplace_back("-Xclang");
@@ -293,9 +296,9 @@ namespace jank::util
     args.emplace_back("-x");
     args.emplace_back("c++-header");
     args.emplace_back("-o");
-    args.emplace_back(output_path.string().c_str());
+    args.emplace_back(output_path_str.c_str());
     args.emplace_back("-c");
-    args.emplace_back(include_path.string().c_str());
+    args.emplace_back(include_path_str.c_str());
     /* We need to add this again for it to get through. Not sure why. */
     args.emplace_back("-std=gnu++20");
 
@@ -310,7 +313,7 @@ namespace jank::util
     }
 
     println(stderr, "done!");
-    return ok(output_path.string().c_str());
+    return ok(output_path_str);
   }
 
   jtl::immutable_string default_target_triple()

